nipype/algorithms/confounds.py

```diff
@@ -402,6 +409,11 @@ class TCompCor(CompCor):
     def _run_interface(self, runtime):
         imgseries = nb.load(self.inputs.realigned_file).get_data()
 
+        if imgseries.ndim != 4:
```

Comment:
The difference is that using numpy requires loading the data into numpy memory, which is expensive. In fact, in ApplyTopUp, it was erroring out. In TCompCor we are loading data into numpy memory anyway, so it is not an additional cost.